Story 806

dianarg opened the pull request:
Resolves #884 
Resolves #874

cmcantalupo committed:
Adds a table of contents for admin tutorial README.md

- Fixes #880.

Change-Id: I89f6593d98ac052b8c817a9bdfa7946fc6d8058e
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dianarg committed:
Add prereqs and reorder sections

Change-Id: Iacdbca48e435a82470057d9759ec0bc666edbbaf
Signed-off-by: Diana Guttman <diana.r.guttman@intel.com>

Commit:
Add integration test of static policy plugin

- Asserts that expected site policy is enforced on non-GEOPM jobs
- Fixes #874

Change-Id: I32b4197323e2814e20ba122ab4f32b6ca2eef604
Signed-off-by: Brandon Baker <brandon.baker@intel.com>

bgeltz committed:
Add prerequisite test script

- Fixes #886.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: I4cefb88380e5fc82a665f17a4d6782b559554553

dianarg committed:
Add documentation and test scripts for some static policy scenarios

- Note: scripts have not been tested.

Change-Id: I729e567145ff7cec28e9abd9d82c3d194b652d72
Signed-off-by: Diana Guttman <diana.r.guttman@intel.com>

cmcantalupo commented on tutorial/admin/README.md:
```diff
@@ -0,0 +1,91 @@
+not using GEOPM will run not have their frequency changed.  The test
+script XXXX can be used to check whether this setup is working.
+
+3. Restrict power limit for jobs that do not use the GEOPM runtime
```
Section header talks about the power limit, but the text uses the EE agent and talks about CPU frequency

cmcantalupo commented on tutorial/admin/setup_fixed_frequency.sh:
```diff
@@ -0,0 +1,70 @@
+
+# Set the GEOPM configuration to use this policy file and the
+# energy efficient agent.
+geopmadmin --magic > $(geopmagent --config-default)
```
This is actually even more magic.  I think the --config-default would be the path the the environment default json.  We really want to populate the policy file that is referred to by the environment default.

dianarg replied:
That should be handled by the geopmagent command above on 51.  We need two json files; what's missing is a command to generate the one below.

cmcantalupo committed:
Small changes to admin readme

Change-Id: If00f6d3ff9bdd03c0cd85e9ec9093304c74a61b2
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Adds geopmadmin command line tool

- Includes a man page for geopmadmin.
- Basic implementation of all features except for check.
- Add an Admin class to enable testing.
- Relates to #891

Change-Id: I52d82efff95b9c9b3d34dd37812ab8d1e9b33793
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dianarg committed:
Add scenarios for override settings and fix scripts

- Add descriptions for scenarios that use the override environment.
- Use geopmadmin tool to get path to config files.
- Fix example setup scripts after testing.
- Add some manual tests to the end of each script; these
  should be handled with automatic integration tests.

Change-Id: If31194d93219b8b069fa6e635da1d54ae50959c7
Signed-off-by: Diana Guttman <diana.r.guttman@intel.com>

cmcantalupo committed:
Add accessor for config path from Environment class

Change-Id: Iaa9b7726bc6b661dda6b98d9a8b8ad9e35461d7a
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Enable whitelist creation for specified architecture

Change-Id: Id95edd974c06eeb8d6e392306565022d303a1790
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add geopm::string_join() helper function

Change-Id: If79fbde1ceca1694baf156eb0c7bf73dbbbbc454
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Complete Admin class implementation and unit tests

- Use geopm::string_join in geopmadmin main
- Clean up geopmadmin_main.cpp and move more
  implementation in to Admin class.
- Fix namespace conflicts for mocked MSRIOGroup sub-objects.
- Get rid of MSRIOGroup object, just use static methods.

Change-Id: I833afb1f2b2a1e0009077a40c96e8029658067d2
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Fixup string overflow issues found by gcc8

- Fix build when mpi is disabled

Change-Id: I84d2d96c3a208e57122a694a1b9165a3a361f594
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

Commit:
Add static member to Environment base class for environment map access

- Refactor EnvironmentImp::parse_environment_file such that it
  is statically accessible.

Change-Id: I09903e7254942f899946a56679e0ea5c8f60558e
Signed-off-by: Brandon Baker <brandon.baker@intel.com>

bgeltz committed:
Remove TURBO_DISENGAGE

Change-Id: I5a0479e41afa8aac2e4316aade98f1793364ba7f
Signed-off-by: Brad Geltz <brad.geltz@intel.com>

dianarg committed:
Add steps to build and install slurm plugin

Change-Id: Iebc72ebbe8d301663bc870d69da6803aba69b3bc
Signed-off-by: Diana Guttman <diana.r.guttman@intel.com>

cmcantalupo committed:
Add missing AdminTest.agent_no_policy and put the rest in ABC order

Change-Id: Ied92cabe7f738b3a573d2ba193a08f2cb8f03220
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dianarg committed:
Add unit tests for helper string_join()

Change-Id: Iaad8fd5d4f4fb807ce1f6d243e00796ab7cfe86f
Signed-off-by: Diana Guttman <diana.r.guttman@intel.com>

bgeltz committed:
Remove is_rapl_working.sh

Change-Id: I3649f442e3b9c1b764574abf90fba854d4001be6
Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add TOC; Cleanup

Change-Id: If968e4d6af86e547353b7e5ba4cc09d28313c88a
Signed-off-by: Brad Geltz <brad.geltz@intel.com>

alawibaba committed:
Revisions to documentation.

alawibaba committed:
Renamed tutorial scripts to reflect section numbering.
